TASVideos/Pages/Account/EmailConfirmationSent.cshtml.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using TASVideos.Core.Services;
+using TASVideos.Core.Services.Email;
+using TASVideos.Data;
+
+namespace TASVideos.Pages.Account;
+
+[AllowAnonymous]
+[IpBanCheck]
+public class EmailConfirmationSentModel : BasePageModel
+{
+	private readonly SignInManager _signInManager;
+	private readonly ApplicationDbContext _db;
+	private readonly IEmailService _emailService;
+
+	public EmailConfirmationSentModel(
+		SignInManager signInManager,
+		ApplicationDbContext db,
+		IEmailService emailService)
+	{
+		_signInManager = signInManager;
+		_db = db;
+		_emailService = emailService;
+	}
+
+	[BindProperty]
+	[Required]
+	[StringLength(256)]
+	[Display(Name = "User Name")]
+	public string UserName { get; set; } = "";
+
+	[BindProperty]
+	[Required]
+	[EmailAddress]
+	[Display(Name = "Email")]
+	public string Email { get; set; } = "";
+	public async Task<IActionResult> OnPost()
+	{
+		if (!await _signInManager.EmailAndUserNameMatch(UserName, Email))
+		{
+			ModelState.AddModelError(nameof(Email), "Username and email entered do not match. Please try again.");
```

Comment:
Unfortunately we can't return a message like this, especially on an unauthenticated page.  It could be used to sniff for emails (though it would be a lot harder with the username matching).  

I think we have no choice but to just return as if the email was sent, and adjust the wording on the page to say something like "if the username and email match, and email will be sent"

Reply:
Done, updated the code so that users can't see whether the validation failed, but they can still know that the button click was received.